Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -166,13 +166,19 @@ public class HTTPClient {
     }
 
     private func resolve(timeout: TimeAmount?, deadline: NIODeadline?) -> TimeAmount? {
-        switch (timeout, deadline) {
+        // compute timeout from deadline
+        var deadlineTimeout: TimeAmount?
+        if case .some(let deadline) = deadline {
+            let now = NIODeadline.now()
+            deadlineTimeout = deadline > now ? deadline - now : .nanoseconds(0)
+        }
+        switch (timeout, deadlineTimeout) {
         case (.some(let timeout), .some(let deadline)):
-            return min(timeout, deadline - .now())
+            return min(timeout, deadline)
         case (.some(let timeout), .none):
             return timeout
         case (.none, .some(let deadline)):
-            return deadline - .now()
+            return deadline
         case (.none, .none):
             return nil
         }
```

Comment:
This mistake seems unnecessary: should this code be using `TimeAmount` at all? It seems like the better choice might be to be using `NIODeadline` everywhere, and turning timeouts in `TimeAmount` form into `NIODeadline`s ASAP.

If we don't do that, we should consider checking for extremely short time-amounts after this call returns (or possibly in this call) and timing them out early rather than making NIO do it.